packages/nodes-base/nodes/Redis/utils.ts

```diff
@@ -16,8 +16,17 @@ export function setupRedisClient(credentials: RedisCredential, isTest = false):
 		port: credentials.port,
 		tls: credentials.ssl === true,
 		connectTimeout: 10000,
-		// Disable reconnection for tests to prevent hanging
-		reconnectStrategy: isTest ? false : undefined,
+		reconnectStrategy: (retries: number, cause: NodeJS.ErrnoException) => {
+			// stop retrying immediately if socket timeout or after 10 retries
+			if (cause.code === 'ECONNREFUSED' || retries >= 10) {
```

Comment:
I don't think we should stop immediately for `ECONNREFUSED`, it could be that the Redis service is down temporarily